### PR TITLE
fix: StablecoinsTable components dark mode text color not see#12648

### PR DESCRIPTION
**Pull request by iepn**
## Describe the bug
There is a text contrast issue on https://ethereum.org/en/stablecoins/ when using dark mode.

In a table on this page when user mouse overs a stablecoin in the table, the text becomes greyed out and hard to read.

CleanShot 2024-04-05 at 02 05 50

## Description

https://github.com/ethereum/ethereum-org-website/assets/57232813/32e48329-2b77-4f21-a68f-3d7f4198c6d2

https://deploy-preview-12666--ethereumorg.netlify.app/en/stablecoins/

## Related Issue

#12648

### CODE REVIEW

**Review by nhsz — CHANGES_REQUESTED** [review]
Thanks! Background color needs to be updated too as it's not using the new DS defined colors cc @nloureiro

**Comment by iepn** [comment]
> Thanks! Background color needs to be updated too as it's not using the new DS defined colors cc @nloureiro

okay

**Comment by iepn** [comment]
I forgot about the new design, sorry! I'll update it shortly.






image

**Review by wackerow — COMMENTED** [review]
Hey @iepn, are all of the changes here necessary for the scope of this issue?

**Comment by iepn** [comment]
> Hey @iepn, are all of the changes here necessary for the scope of this issue?


Hello, if the issue is solely with the background color not displaying correctly in dark mode, then the original submission would suffice. However, upon the reminder from @nhsz, I reviewed the Figma design system of ethereum.org and found differences between the design system and the current implementation. Therefore, I applied the styles of the new table design system.

Please let me know if this is not appropriate, and I will revert it.

**Comment by wackerow** [comment]
Ah okay, sounds good... apologies, had missed that message. Will review again from that scope

File: src/components/StablecoinsTable.tsx [hunk 1]
```diff
@@ -50,20 +50,18 @@ const StablecoinsTable = ({
...
       my={8}
-      borderRadius="sm"
-      border={`1px solid ${textColor}`}
       bg="background.base"
       mb={8}
       minW="720px"
     >
-      <Thead bg="ednBackground" color="text200">
-        <Tr borderBottom={`1px solid ${textColor}`} mb="1px">
+      <Thead bg="background.highlight" color="text200">
```
Comment (wackerow) [inline src/components/StablecoinsTable.tsx:57]: cc: @nloureiro okay switching `text200` (light: #666666, dark: #b2b2b2) to `body.medium` (light: gray.500 or #616161, dark: gray.400 or #8C8C8C)?

```suggestion
      <Thead bg="background.highlight" color="body.medium">
```

File: src/components/StablecoinsTable.tsx [hunk 2]
```diff
@@ -50,20 +50,18 @@ const StablecoinsTable = ({
...
+        <Tr mb="1px">
```
Comment (wackerow) [inline src/components/StablecoinsTable.tsx:58]: Could someone drop the link to the Figma for this? Feel like we shouldn't have 1px adjustments

File: src/components/StablecoinsTable.tsx [hunk 3]
```diff
@@ -90,30 +88,26 @@ const StablecoinsTable = ({
...
                 <>{name}</>
               </Flex>
             </Td>
             <Td>
-              <Flex align="center">{marketCap}</Flex>
+              <Flex align="center" pt={2}>{marketCap}</Flex>
             </Td>
             <Td>
-              <Flex align="center">{stablecoinsType[type]}</Flex>
+              <Flex align="center" pt={2}>{stablecoinsType[type]}</Flex>
```
Comment (wackerow) [inline src/components/StablecoinsTable.tsx:110]: Would need to double check with the actual Figma, but would advocate that we not rely on padding-top for positioning these. @nloureiro is this small adjustment desired?

If we're adding `pt="2"` to these, we should likely ditch that and add 1 to the y-padding of the parent `Td` which currently has `p="4"`... adding `py="5"` would do this, but again, using 5 is a bit odd, and we may want to consider simplifying.

**Review by wackerow — COMMENTED** [review]
cc: @nloureiro

File: src/components/StablecoinsTable.tsx (hunk 2 above)
Comment (iepn) [reply src/components/StablecoinsTable.tsx:58]: > Could someone drop the link to the Figma for this? Feel like we shouldn't have 1px adjustments

https://www.figma.com/file/NrNxGjBL0Yl1PrNrOT8G2B/ethereum.org-Design-System?type=design&node-id=1961%3A33521&mode=design&t=cm7YrGNCoOVawgRV-1

File: src/components/StablecoinsTable.tsx (hunk 3 above)
Comment (iepn) [reply src/components/StablecoinsTable.tsx:110]: I used pt 2 here to try to horizontally center the content as much as possible, but please give me some time, I will come up with a better solution in the next submission.

Comment (iepn) [reply src/components/StablecoinsTable.tsx:110]: > Would need to double check with the actual Figma, but would advocate that we not rely on padding-top for positioning these. @nloureiro is this small adjustment desired?
> 
> If we're adding `pt="2"` to these, we should likely ditch that and add 1 to the y-padding of the parent `Td` which currently has `p="4"`... adding `py="5"` would do this, but again, using 5 is a bit odd, and we may want to consider simplifying.

I used pt 2 here to try to horizontally center the content as much as possible, but please give me some time, I will come up with a better solution in the next submission.

File: src/components/StablecoinsTable.tsx (hunk 1 above)
Comment (iepn) [reply src/components/StablecoinsTable.tsx:57]: > cc: @nloureiro okay switching `text200` (light: #666666, dark: #b2b2b2) to `body.medium` (light: gray.500 or #616161, dark: gray.400 or #8C8C8C)?

I'm very sorry, I overlooked `text200`. Thank you for pointing it out, I appreciate your help.

File: src/components/StablecoinsTable.tsx (hunk 3 above)
Comment (iepn) [reply src/components/StablecoinsTable.tsx:110]: use `verticalAlign="inherit"` ！

**Comment by iepn** [comment]
You will absolutely love the code in this commit! https://github.com/ethereum/ethereum-org-website/pull/12666/commits/85d573cfe8cf2cb511f17f4d3baeebb70fd98795

https://deploy-preview-12666--ethereumorg.netlify.app/en/stablecoins/

File: src/components/StablecoinsTable.tsx [hunk 4]
```diff
@@ -47,23 +47,15 @@ const StablecoinsTable = ({
...
+      <Thead bg="background.highlight" color="body.medium">
+        <Tr>
           {columns.map((column, idx) => (
             <Th
               key={idx}
-              py={5}
+              fontWeight="bold"
               fontSize="md"
-              fontWeight="normal"
+              verticalAlign="inherit"
```
Comment (wackerow) [inline src/components/StablecoinsTable.tsx:58]: Do we need this?
```suggestion
```

**Review by wackerow — CHANGES_REQUESTED** [review]
Nice! Yeah, this looks more like our DS table... I see the middle vertical positioning which makes sense here give the icons, but left a note about the inherited property... I think we can cut that.

Otherwise lgtm!

File: src/components/StablecoinsTable.tsx (hunk 4 above)
Comment (iepn) [reply src/components/StablecoinsTable.tsx:58]: Trust me, it's absolutely necessary. Without it, the text won't be centered anymore.

Comment (iepn) [reply src/components/StablecoinsTable.tsx:58]: image

I've made a diagram where you can clearly see the difference between having "verticalAlign="inherit"" and not having it.

cc @wackerow

**Review by nloureiro — APPROVED** [review]

**Comment by wackerow** [comment]
@all-contributors please add @iepn for code

**Pull request merged**